src/v2/context.in.rs

```diff
@@ -32,6 +32,21 @@ impl Context {
             Context::Dir(Path::new(&s_ref).to_owned())
         }
     }
+
+    /// Determines if two Contexts are equivalent. This is only true
+    /// if they are the same directory path, or if they are same git
+    /// repository and branch. Subdirectories of a repository may differ
+    pub fn is_equivalent_to(&self, other: &Context) -> bool {
+        match (self, other) {
+            (&Context::Dir(_), &Context::GitUrl(_)) => false,
+            (&Context::GitUrl(_), &Context::Dir(_)) => false,
+            (&Context::Dir(ref dir_1), &Context::Dir(ref dir_2)) => dir_1 == dir_2,
+            (&Context::GitUrl(ref git_url_1), &Context::GitUrl(ref git_url_2)) => {
+                git_url_1.repository() == git_url_2.repository() &&
+                    git_url_1.branch() == git_url_2.branch()
```

Comment:
The `GitUrl`-specific portion of this function should probably be moved to `GitUrl`, with a name like `can_share_checkout_with`.

Reply:
👍